src/blar_graph/graph_construction/utils/tree_parser.py

```diff
@@ -45,56 +26,107 @@ def get_function_name(call_str):
+        elif (type == "object" or type == "method") and nested_object:
+            continue
+        else:
+            decomposed_calls.append(decompose_node.text.decode())
+
+    return decomposed_calls
+
+
+def get_function_calls(node: Node, assigments_dict: dict, language: str) -> list[str]:
     code_text = node.text
+    function_calls = []
 
     parser = tree_sitter_languages.get_parser(language)
     tree = parser.parse(bytes(code_text, "utf-8"))
-    node_names = map(lambda node: node, traverse_tree(tree))
+    language = tree_sitter_languages.get_language(language)
 
-    function_calls = []
+    assignment_query = language.query("""(assignment left: _ @variable right: _ @expression)""")
 
-    for tree_node in node_names:
-        if tree_node.type == "expression_statement":
-            statement_children = tree_node.children
-            if statement_children[0].type == "assignment":
-                assigment = statement_children[0].named_children
-
-                variable_identifier = assigment[0]
-                assign_value = assigment[1]
-                if assign_value.type == "call":
-                    expression = assign_value
-                    expression_identifier = expression.named_children[0].text.decode()
-
-                    assigments_dict[variable_identifier.text.decode("utf-8")] = expression_identifier
-
-        if tree_node.type == "call":
-            call_children = tree_node.named_children
-            if call_children[0].type == "attribute" and call_children[1].type == "argument_list":
-                attribute_children = call_children[0].named_children
-                root_caller = attribute_children[0]
-                if root_caller.type == "identifier":
-                    root_caller_identifier = root_caller.text.decode("utf-8")
-                    if root_caller_identifier in assigments_dict:
-                        function_calls.append(
-                            assigments_dict[root_caller_identifier]
-                            + "."
-                            + attribute_children[1].text.decode("utf-8")
-                            + "()"
-                        )
-                        continue
-            function_calls.append(tree_node.text.decode("utf-8"))
-
-    parsed_function_calls = map(
-        lambda x: parse_function_call(x, node.metadata["inclusive_scopes"]),
-        function_calls,
-    )
+    assigments = assignment_query.captures(tree.root_node)
+
+    for assigment_node, assigment_type in assigments:
+        if assigment_type == "variable":
+            variable_identifier_node = assigment_node
+            continue
+        if assigment_type == "expression":
+            assign_value = assigment_node
+
+        if assign_value.type == "call":
+            expression = assign_value
+            expression_identifier = expression.named_children[0].text.decode()
+            variable_identifier = variable_identifier_node.text.decode()
```

Comment:
shouldn't this be indented in case assign_value = None?
```suggestion
          if assign_value.type == "call":
              expression = assign_value
              expression_identifier = expression.named_children[0].text.decode()
              variable_identifier = variable_identifier_node.text.decode()
```